adapters/rack/lib/opentelemetry/adapters/rack/middlewares/tracer_middleware.rb

```diff
@@ -0,0 +1,193 @@
+          class << self
+            def allowed_rack_request_headers
+              @allowed_rack_request_headers ||= Array(config[:allowed_request_headers]).each_with_object({}) do |header, memo|
+                memo["HTTP_#{header.to_s.upcase.gsub(/[-\s]/, '_')}"] = build_attribute_name('http.request.headers.', header)
+              end
+            end
+
+            def allowed_response_headers
+              @allowed_response_headers ||= Array(config[:allowed_response_headers]).each_with_object({}) do |header, memo|
+                memo[header] = build_attribute_name('http.response.headers.', header)
+                memo[header.to_s.upcase] = build_attribute_name('http.response.headers.', header)
+              end
+            end
+
+            def build_attribute_name(prefix, suffix)
+              prefix + suffix.to_s.downcase.gsub(/[-\s]/, '_')
+            end
+
+            def config
+              Rack::Adapter.instance.config
+            end
+
+            private
+
+            def clear_cached_config
+              @allowed_rack_request_headers = nil
+              @allowed_response_headers = nil
+            end
+          end
+
+          EMPTY_HASH = {}.freeze
+
+          def initialize(app)
+            @app = app
+          end
+
+          def call(env)
+            original_env = env.dup
+            extracted_context = OpenTelemetry.propagation.extract(env)
+            frontend_context = create_frontend_span(env, extracted_context)
+
+            # restore extracted context in this process:
+            OpenTelemetry::Context.with_current(frontend_context || extracted_context) do
+              request_span_name = create_request_span_name(env['REQUEST_URI'] || original_env['PATH_INFO'])
+              tracer.in_span(request_span_name,
+                             attributes: request_span_attributes(env: env),
+                             kind: :server) do |request_span|
+                @app.call(env).tap do |status, headers, response|
+                  set_attributes_after_request(request_span, status, headers, response)
+                end
+              end
+            end
+          ensure
+            finish_span(frontend_context)
+          end
+
+          private
+
+          # return Context with the frontend span as the current span
+          def create_frontend_span(env, extracted_context)
+            request_start_time = OpenTelemetry::Adapters::Rack::Util::QueueTime.get_request_start(env)
+
+            return unless config[:record_frontend_span] && !request_start_time.nil?
+
+            span = tracer.start_span('http_server.queue',
+                                     with_parent_context: extracted_context,
+                                     attributes: {
+                                       'component' => 'http',
+                                       'service' => config[:web_service_name],
+                                       'start_time' => request_start_time
```

Comment:
One more issue. This needs to be a number, not a timestamp.

```suggestion
                                       'start_time' => request_start_time.to_f
```